cmd/soroban-cli/src/commands/keys/generate.rs

```diff
@@ -60,17 +102,31 @@ impl Cmd {
         } else {
             seed_phrase
         };
-        self.config_locator.write_identity(&self.name, &secret)?;
-        if !self.no_fund {
-            let addr = secret.public_key(self.hd_path)?;
-            let network = self.network.get(&self.config_locator)?;
-            network
-                .fund_address(&addr)
-                .await
-                .map_err(|e| {
-                    tracing::warn!("fund_address failed: {e}");
-                })
-                .unwrap_or_default();
+
+        // Check if identity exists
+        let Ok(existing_secret) = self.config_locator.read_identity(&self.name) else {
+            // Identity doesn't exist, create new one
+            self.write_and_fund_identity(&secret).await?;
+            return Ok(());
+        };
+        if self.seed.is_some() || self.default_seed {
+            // Compare secrets only if seed is provided
+            match (
+                existing_secret.private_key(self.hd_path),
+                secret.private_key(self.hd_path),
+            ) {
+                (Ok(existing_pk), Ok(new_pk)) if existing_pk == new_pk => {
+                    self.handle_existing_identity(&existing_secret).await?;
+                }
+                _ => {
+                    // Secrets don't match
+                    eprintln!("An identity with the name {} already exists but has a different secret. Overwriting...", self.name);
+                    self.write_and_fund_identity(&secret).await?;
```

Comment:
I think the idea that the command will implicitly overwrite in some edge cases is surprising. If we're saying "overwrite by default is a bug" then it's a bug that it still overwrites here.

The only time overwriting should be allowed is if an explicit `--overwrite` is provided so that the overwrite only occurs explicitly.

Reply:
I added an `--overwrite` flag which simplified the logic as there was no need to compare existing secrets.